02_knight_and_princess/dana_stepina/main.cpp

```diff
@@ -0,0 +1,159 @@
+  std::vector<Edge> edges_;
+  EdgeId edge_id_counter_ = INVALID_ID;
+  EdgeId vertex_id_counter_ = INVALID_ID;
+  void add_vertices_connection(const VertexId& from_vertex_id,
+                               const VertexId& to_vertex_id,
+                               const EdgeId& edge);
+
+  EdgeId get_new_edge_id() { return ++edge_id_counter_; }
+  EdgeId get_new_vertex_id() { return ++vertex_id_counter_; }
+};
+
+void Graph::add_vertex() {
+  vertices_.emplace_back(get_new_vertex_id());
+}
+void Graph::add_edge(const VertexId& from_vertex_id,
+                     const VertexId& to_vertex_id) {
+  assert(!has_vertex(from_vertex_id) &&
+         "There is no such vertex in the graph.");
+  assert(!has_vertex(to_vertex_id) && "There is no such vertex in the graph.");
+
+  edges_.emplace_back(get_new_edge_id(), from_vertex_id, to_vertex_id);
+  Edge new_edge = edges_.back();
+  add_vertices_connection(from_vertex_id, to_vertex_id, new_edge.id);
+}
+void Graph::add_vertices_connection(const VertexId& from_vertex_id,
+                                    const VertexId& to_vertex_id,
+                                    const EdgeId& edge) {
+  vertices_[from_vertex_id].add_edge_id(edge);
+  vertices_[to_vertex_id].add_edge_id(edge);
+}
+
+//Вывод в файл json
+std::string get_vertex_string(const Vertex& vertex) {
+  std::string str_vertex =
+      "\t\t{\"id\":" + std::to_string(vertex.id) + ",\"edge_ids\":[";
+  for (const auto& edge_id : vertex.get_edge_ids()) {
+    str_vertex += std::to_string(edge_id);
+    ((edge_id != vertex.get_edge_ids().back()) ? str_vertex += ","
+                                               : str_vertex += "]}");
+  }
+  return str_vertex;
+}
+std::string get_edge_string(const Edge& edge) {
+  std::string str_edge = "\t\t{\"id\":" + std::to_string(edge.id) +
+                         ",\"vertex_ids\":[" +
+                         std::to_string(edge.vertex_start) + "," +
+                         std::to_string(edge.vertex_end) + "]}";
+  return str_edge;
+}
+std::string get_graph_string(const Graph& graph) {
+  std::string str_graph = "{\n\t\"vertices\": [\n";
+
+  // write vertices
+  for (const auto& vertex : graph.get_vertices()) {
+    ((vertex.id != graph.get_vertices().size() - 1)
+         ? str_graph += get_vertex_string(vertex) + ",\n"
+         : str_graph += get_vertex_string(vertex) + "\n");
+  }
+
+  // write edges
+  str_graph += "\t\t],\n\t\"edges\": [\n";
+  for (const auto& edge : graph.get_edges()) {
+    ((edge.id != graph.get_edges().back().id)
+         ? str_graph += get_edge_string(edge) + ",\n"
+         : str_graph += get_edge_string(edge) + "\n");
+  }
+
+  str_graph += "\t\t]\n}\n";
+  return str_graph;
+}
+void write_graph_json_file(const Graph& graph) {
+  std::ofstream out("graph.json");
+  out << get_graph_string(graph);
+  out.close();
+}
+
+int main() {
+  // GRAPH
+  const int vertex_count = 14;
+  const std::vector<std::array<int, 2>> vertex_connections = {

```

Comment:
Будет логичнее использовать `std::pair`
```suggestion
  const std::vector<std::pair<int>> vertex_connections = {
```